compiler/testData/codegen/box/localClasses/anonymousObjectInInitializer.kt

```diff
@@ -1,3 +1,4 @@
+// IGNORE_BACKEND_FIR: JVM_IR
```

Comment:
This one has an unrelated, unresolved call, which is now visible to IR backend. :\

Reply:
Which one is it, and if it was not present before, why was the test not failing?

Reply:
It was `toString` call with the explicit receiver of anonymous object. I thought that unresolved call was already there, but it turned out that it's due to this change. After investigating when/how the resolution result has been changed, found that indeed the initial change bothered that `toString` to be transformed by `TYPE`-phase transformer; then its return type is not resolved; then from the call site, call completer gave up, since the candidate still has non-resolved type (user-defined type in this example).

On the bright side, by comparing resolution differences before/after the initial change against the motivation test _and_ this regression, I learned about those resolution phases, phase-based transformers, etc., and found that there is a unified place to transform declarations inside local classes. That means, the declarations in question (with `unknown` visibility) were somehow not transformed, and the main reason was the containing class was determined as non-local if it's inside anonymous initializer.

That led me to find where such (should-be-)`local` class was incorrectly determined, and concluded that the current scope-based logic wasn't a good fit. Will update the top-level comment to explain that.